0000-ilp-over-http.md

```diff
@@ -0,0 +1,81 @@
+| `<body>` | Binary, Maximum of 32767 Bytes | N | End-to-end data used by Transport Layer protocols |
+| `ILP-...` | UTF-8 String | N | Additional headers that MUST be forwarded by all connectors, even if they are not understood |
+| `ILP-Destination-Amount` | Unsigned 64-Bit Integer | N | _(OPTIONAL)_ The amount the connectors should attempt to deliver to the destination account. If unspecified, connectors will forward the payment to the destination by applying their local rate. Not all connectors will support delivery so Transport Layer protocols that set this field SHOULD accept overpayment. |
+
+### Response
+
+#### Success
+
+```http
+HTTP/1.1 200 OK
+ILP-Fulfillment: cz/9RGv1PVjhKIOoyPvWkAs8KrBpIJh8UrYsQ8j34CQ=
+
+<body>
+```
+
+| Field | Type | Modified at Each Hop? | Description |
+|---|---|---|---|
+| `ILP-Fulfillment` | Base64-Encoded String, 32 Bytes | N | Preimage of the `ILP-Condition` |
+| `<body>` | Binary, Maximum of 32767 Bytes | N | End-to-end data used by Transport Layer protocols |
+| `ILP-...` | UTF-8 String | N | Additional headers that MUST be forwarded by all connectors, even if they are not understood |
+
+#### Error
+
+HTTP Error Codes MAY be used to indicate certain types of failures, but [ILP Error Codes](./0003-interledger-protocol/0003-interledger-protocol.md#ilp-error-codes) MUST be used and relayed by connectors.
+
+```http
+HTTP/1.1 404 Not Found
+ILP-Error-Code: F02
+ILP-Error-Name: Unreachable
+ILP-Error-Triggered-By: g.usd.acmebank.user12345.xclkv-909sdf
+ILP-Error-Triggered-At: 2017-12-07T19:11:21.917Z
+ILP-Error-Forwarded-By: g.usd.acmebank.connector,g.crypto.bitcoin.1A1zP1eP5QGefi2DMPTfTL5SLmv7DivfNa
+
+<body>
+```
+
+| Field | Type | Modified at Each Hop? | Description |
+|---|---|---|---|
+| `ILP-Error-Code` | [ILP Error Code](./0003-interledger-protocol/0003-interledger-protocol.md#ilp-error-codes) | N | 3-letter code identifying the error |
+| `ILP-Error-Name` | String | N | Human-readable name corresponding to the `ILP-Error-Code` |
+| `ILP-Error-Triggered-By` | ILP Address | N | Address of the party that initially emitted the error |
+| `ILP-Error-Triggered-At` | ISO 8601 Timestamp in UTC | N | Time when the error was initially emitted |
+| `ILP-Error-Forwarded-By` | Comma-Separated List of ILP Addresses | Y | List of connectors that relayed this error. Connectors SHOULD append their ILP addresses to the end of this list when relaying the error |
+| `<body>` | Binary, Maximum of 32767 Bytes | N | End-to-end data used by Transport Layer Protocols |
+| `ILP-...` | UTF-8 String | N | Additional headers that MUST be forwarded by all connectors, even if they are not understood |
```

Comment:
Note that to support both a response body and the additional headers, we would probably want to add a `data` property to the [LPIv2](https://github.com/interledger/rfcs/pull/347/files#diff-94ebb08ebfc42105702c38dc2b373f2dR384) error, in addition to the `additionalInfo`.